kong/tools/queue_schema.lua

```diff
@@ -0,0 +1,42 @@
+local Schema = require "kong.db.schema"
+
+-- TODO: enable the descriptions once they are accepted in Schemas
+return Schema.define {
+  type = "record",
+  fields = {
+    { name = {
+      type = "string",
+      -- description = "name of the queue",
+    } },
+    { batch_max_size = {
+      type = "number",
+      default = 1,
+      -- description = "maximum number of entries to be that can be processed at a time"
+    } },
+    { max_delay = {
+      type = "number",
+      default = 1,
+      -- description = "maximum number of (fractional) seconds to elapse after the first entry was queued before the queue starts calling the handler",
+    } },
+    { capacity = {
+      type = "number",
+      default = 10000,
+      -- description = "maximum number of entries that can be waiting on the queue",
+    } },
+    { string_capacity = {
+      type = "number",
+      default = nil,
+      -- description = "maximum number of bytes that can be waiting on a queue, requires string content",
```

Comment:
How does a user know if a plugin is using string content or not?

Reply:
This must be documented in the plugin that uses the queue.

Reply:
`string_capacity` is now called `max_bytes` as well.

At this point, only the `http-log` plugin puts strings onto the queue, so the `max_bytes` parameter will only be documented for that plugin.  It will still be in the shared schema, though.